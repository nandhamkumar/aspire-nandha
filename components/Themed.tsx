// Nandha - Added additional themed components
// This helps to style the components as per the device theme - Dark/Light

import { Text as DefaultText, View as DefaultView, TouchableOpacity as DefaultTouchableOpacity, TextInput as DefaultTextInput, KeyboardTypeOptions } from 'react-native';
import { Ionicons as DefaultIonicons } from '@expo/vector-icons';
import Colors from '../constants/Colors';
import useColorScheme from '../hooks/useColorScheme';


export function useThemeColor(
  props: { light?: string; dark?: string },
  colorName: keyof typeof Colors.light & keyof typeof Colors.dark
) {
  const theme = useColorScheme();
  const colorFromProps = props[theme];

  if (colorFromProps) {
    return colorFromProps;
  } else {
    return Colors[theme][colorName];
  }
}

type ThemeProps = {
  lightColor?: string;
  darkColor?: string;
};

// Nandha - Created custom properties that are used by our components
type CustomProps = {
  header?: string;
  title?: string;
  onSwitch?: any;
  icon?: string;
  switchValue?: boolean;
  goBack?: any;
  onPress?: any;
  value?: string | number;
  text?: string;
  onChangeText?: any;
  placeholder?: string;
  detail?: string;
  keyboard?: KeyboardTypeOptions | undefined;
  name?: any;
  size?: number;
};

type CustomTextProps = {
  placeholder?: string;
  keyboard?: KeyboardTypeOptions | undefined;
  value?: string | undefined;
  onChangeText?: any;
}

export type TextProps = CustomTextProps & ThemeProps & DefaultText['props'];
export type ViewProps = CustomProps & ThemeProps & DefaultView['props'];

export function Text(props: TextProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const color = useThemeColor({ light: lightColor, dark: darkColor }, 'text');

  return <DefaultText style={[{ color }, style]} {...otherProps} />;
}

export function View(props: ViewProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, 'background');

  return <DefaultView style={[{ backgroundColor }, style]} {...otherProps} />;
}


// Nandha - Auto selects color based on the device theme
export function Ionicons(props: ViewProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const color = useThemeColor({ dark: darkColor, light: lightColor }, 'icon');
  return <DefaultIonicons color={color} style={[style]} {...otherProps} />;
}

// Nandha - Auto selects color based on the device theme
export function TouchableOpacity(props: ViewProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, 'background');

  return <DefaultTouchableOpacity style={[{ backgroundColor }, style]} {...otherProps} />;
}


// Nandha - Auto selects color based on the device theme and style the Input Text as per our theme
export function TextInput(props: TextProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const color = useThemeColor({ light: lightColor, dark: darkColor }, 'text');

  return <DefaultTextInput keyboardType={props.keyboard} style={[{ color, fontSize: 20, fontWeight: 'bold' }, style]} {...otherProps} />

}