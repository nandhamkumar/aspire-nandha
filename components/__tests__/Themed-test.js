// Author: Nandha
// Unit Test Themed Components - Snapshot Testing

import * as React from 'react';
import renderer from 'react-test-renderer';
import { Ionicons, Text, TouchableOpacity, View } from '../Themed';


//Test Text Component
it(`Text renders correctly`, async() => {
  const tree = renderer.create(<Text>Snapshot Test</Text>).toJSON();

  expect(tree).toMatchSnapshot();
});


//Test View Component
it(`View renders correctly`, async() => {
  const tree = renderer.create(<View><Text>Snapshot Test</Text></View>).toJSON();

  expect(tree).toMatchSnapshot();
});


//Test Ionicons Component
it(`Ionicons renders correctly`, async() => {
  const tree = renderer.create(<Ionicons ></Ionicons>).toJSON();

  expect(tree).toMatchSnapshot();
});

//Test TouchableOpacity Component
it(`TouchableOpacity renders correctly`, async() => {
  const tree = renderer.create(<TouchableOpacity><Text>Snapshot Test</Text></TouchableOpacity>).toJSON();

  expect(tree).toMatchSnapshot();
});
