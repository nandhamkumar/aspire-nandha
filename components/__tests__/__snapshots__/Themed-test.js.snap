// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ionicons renders correctly 1`] = `<Text />`;

exports[`Text renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
      },
      undefined,
    ]
  }
>
  Snapshot Test
</Text>
`;

exports[`TouchableOpacity renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        undefined,
      ]
    }
  >
    Snapshot Test
  </Text>
</View>
`;

exports[`View renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        undefined,
      ]
    }
  >
    Snapshot Test
  </Text>
</View>
`;
